src/mapeo-manager.js

```diff
@@ -0,0 +1,148 @@
+import { randomBytes } from 'crypto'
+import path from 'path'
+import { KeyManager } from '@mapeo/crypto'
+import Database from 'better-sqlite3'
+import { eq } from 'drizzle-orm'
+import { drizzle } from 'drizzle-orm/better-sqlite3'
+import { migrate } from 'drizzle-orm/better-sqlite3/migrator'
+import { IndexWriter } from './index-writer/index.js'
+import { MapeoProject } from './mapeo-project.js'
+import { projectKeysTable, projectTable } from './schema/client.js'
+import { ProjectKeys } from './generated/keys.js'
+
+const CLIENT_SQLITE_FILE_NAME = 'client.db'
+
+export class MapeoManager {
+  #keyManager
+  #projectSettingsIndexWriter
+  #storagePath
+  #db
+  /** @type {Map<string, MapeoProject>} */
+  #activeProjects
+
+  /**
+   * @param {Object} [opts]
+   * @param {string} [opts.storagePath] Folder for all data storage (hypercores and sqlite db). Folder must exist. If not defined, everything is stored in-memory
+   */
+  constructor({ storagePath } = {}) {
+    const dbPath =
+      storagePath !== undefined
+        ? path.join(storagePath, CLIENT_SQLITE_FILE_NAME)
+        : ':memory:'
+
+    const sqlite = new Database(dbPath)
+    this.#db = drizzle(sqlite)
+    migrate(this.#db, { migrationsFolder: './drizzle/client' })
+
+    this.#keyManager = new KeyManager(KeyManager.generateRootKey())
```

Comment:
`rootKey` should be a constructor parameter. Otherwise we have no way of recreating this instance for tests (or actual use)

Reply:
I think required is best.

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/199/commits/f4d60c5327c468407190304c4cda13d28ff0fbd5